ledgerManager.php

```diff
@@ -0,0 +1,319 @@
+<?php
+
+class LedgerManager {
+	protected $pdo = null;
+	protected $table = null;
+	protected $overallBalance = null;
+
+	/**
+	 * @param string $username
+	 * @param string $password
+	 * @param string $databae
+	 * @param string|null $table
+	 */
+	public function __construct($username, $password, $database, $table = 'ledger')
+	{
+		$this->pdo = new PDO("mysql:host=localhost;dbname=". $database, $username, $password);
+		$this->table = $table;
+	}
+
+	/**
+	 * @param int $transactionId
+	 * @param string $description
+	 * @param float $amount
+	 * @param bool $cleared
+	 * @return bool
+	 */
+	public function updateTransaction($transactionId, $description, $amount, $cleared)
+	{
+		$query = $this->pdo->prepare("
+			update {$this->table}
+			set description = :description,
+			amount = :amount,
+			cleared = :cleared
+			where id = :transactionId
+		");
+		return $query->execute([
+			':description' => $description,
+			':amount' => $amount,
+			':cleared' => $cleared,
+			':transactionId' => $transactionId
+		]);
+	}
+
+	/**
+	 * @param string $description
+	 * @param float $amount
+	 * @param bool $cleared
+	 * @return bool
+	 */
+	public function insertTransaction($description, $amount, $credit)
+	{
+		$query = $this->pdo->prepare("
+			insert into {$this->table}
+			(description, amount, credit)
+			values (:description, :amount, :credit)
+		");
+		return $query->execute([
+			':description' => $description,
+			':amount' => $amount,
+			':credit' => $credit,
+		]);
+	}
+
+	/**
+	 * @param int $transactionId
+	 * @return bool
+	 */
+	public function deleteTransaction($transactionId)
+	{
+		$query = $this->pdo->prepare("
+			delete from {$this->table}
+			where id = :transactionId
+		");
+		return $query->execute([':transactionId' => $transactionId]);
+	}
+
+	/**
+	 * This method returns the sum of all credit or debit transactions
+	 * for (an optional) date range
+	 *
+	 * @param bool $credit
+	 * @param string|null $startDate
+	 * @param string|null $endDate
+	 * @return float
+	 */
+	protected function getTotalSumCreditTransactions($credit = true, $startDate = null, $endDate = null)
```

Comment:
i don't like the name of this method.  could be broken up into 2 wrapper methods: getTotalSumCreditTransactions and getTotalSumDebitTransactions, while this is renamed as getSumOfLikeTransactionsOverARange...or something more betterer.